corehq/apps/export/static/export/js/datasource_export.js

```diff
@@ -0,0 +1,26 @@
+hqDefine("export/js/datasource_export",[
+    'jquery',
+    'knockout',
+], function ($, ko) {
+
+    function datasourceExportViewModel() {
+        'use strict';
+        var self = {};
+
+        self.dataSource = ko.observable();
+        self.haveDatasources = ko.computed(function () {
+            if (self.dataSource() === undefined) { return false; }
+            return self.dataSource().length > 0;
+        });
+        self.updateExportButton = ko.computed(function () {
+            $("#datasources_export_submit_button_id").prop('disabled', !self.haveDatasources());
```

Comment:
This can be replaced with [enable/disable KO binding](https://knockoutjs.com/documentation/enable-binding.html) directly on the button itself.